src/listpack.c

```diff
@@ -3050,6 +3099,38 @@ int listpackTest(int argc, char *argv[], int flags) {
         lpFree(lp);
     }
 
+    TEST("Test lpFindInteger") {
+        /* Create a listpack with a mix of integers and strings */
+        unsigned char *lp = lpNew(0);
+        lp = lpAppendInteger(lp, 100);
+        lp = lpAppend(lp, (unsigned char*)"hello", 5);
+        lp = lpAppendInteger(lp, -50);
+        unsigned char *p;
+        long long val;
+
+        /* Test: Find an existing integer (100) */
+        p = lpFindInteger(lp, NULL, 100, 0);
+        assert(p != NULL);
+        lpGetIntegerValue(p, &val);
+        assert(val == 100);
+
+        /* Test: Find an existing integer (-50) */
+        p = lpFindInteger(lp, NULL, -50, 0);
+        assert(p != NULL);
+        lpGetIntegerValue(p, &val);
+        assert(val == -50);
+
+        /* Test: Skip occurrences - Append a duplicate and skip the first occurrence */
+        lp = lpAppendInteger(lp, 100);
+        /* Here, 'skip' is set to 1 to skip the first occurrence of 100 */
+        p = lpFindInteger(lp, NULL, 100, 1);
```

Comment:
please add a test to verify that -50 can be found
and adding another numeric element, and then use `skip` to skip it.